### PR TITLE
ENH use classification instead of regression in random forests

### LINKED ISSUE #312: Introduce random forest classifier instead of random forest regressor
After some discussion with @ogrisel we think it makes things easier to introduce `RandomForestClassifier` instead of `RandomForestRegressor`.

### PR DESCRIPTION
**Pull request by glemaitre**
closes #312

### CODE REVIEW

File: python_scripts/ensemble_random_forest.py [hunk 1]
```diff
@@ -109,10 +110,12 @@
 
 # %% [markdown]
 #
+# Not that the performance of the bagged trees is already much better than the
```
Comment (glemaitre) [inline python_scripts/ensemble_random_forest.py:113]: Not -> Note

File: python_scripts/ensemble_random_forest.py [hunk 2]
```diff
@@ -109,10 +110,12 @@
...
+# performance of a single tree.
```
Comment (glemaitre) [inline python_scripts/ensemble_random_forest.py:114]: there is a performance here

File: python_scripts/ensemble_random_forest.py [hunk 3]
```diff
@@ -149,19 +152,28 @@
...
 # default values reflect good practices given in the scientific literature.
-# However, `max_features` is one of the hyperparameter to consider when tuning
-# a random forests.
 #
-# Note that bagging ensemble exposes a parameter `max_features` as well.
-# However, the process is different since the base estimator is not necessarily
-# a decision tree. Indeed, the features subsampling take place at the model
-# level by selecting a subset of feature on each bootstrap sample.
+# However, `max_features` is one of the hyperparameters to consider when tuning
+# a random forests:
```
Comment (glemaitre) [inline python_scripts/ensemble_random_forest.py:157]: random forests -> random forest

File: python_scripts/ensemble_random_forest.py [hunk 4]
```diff
@@ -149,19 +152,28 @@
...
+# - too much randomness in the trees can lead to underfitting base models and
```
Comment (glemaitre) [inline python_scripts/ensemble_random_forest.py:158]: underfitting -> underfitted

File: python_scripts/ensemble_random_forest.py [hunk 5]
```diff
@@ -149,19 +152,28 @@
...
-# | `BaggingRegressor`       | User specified (flexible) | `n_features` (no subsampling)    | Model level                   |
-# | `RandomForestRegressor`  | `DecisionTreeRegressor`   | `n_features` (no subsampling)    | Tree node level               |
+# | Ensemble model class     | Base model class          | Default value for `max_features`   | Features subsampling strategy |
+# |--------------------------|---------------------------|------------------------------------|-------------------------------|
+# | `BaggingClassifier`      | User specified (flexible) | `n_features` (no&nbsp;subsampling) | Model level                   |
+# | `RandomForestClassifier` | `DecisionTreeClassifier`  | `sqrt(n_features)`                 | Tree node level               |
+# | `BaggingRegressor`       | User specified (flexible) | `n_features` (no&nbsp;subsampling) | Model level                   |
+# | `RandomForestRegressor`  | `DecisionTreeRegressor`   | `n_features` (no&nbsp;subsampling) | Tree node level               |
+
+# %%
```
Comment (glemaitre) [inline python_scripts/ensemble_random_forest.py:179]: No need for this one

File: python_scripts/ensemble_random_forest.py (hunk 4 above)
Comment (ogrisel) [reply python_scripts/ensemble_random_forest.py:158]: For the record Wikipedia says "underfitted" which I believed was not correct English but so be it.

**Pull request merged**